tests/datasource/fluent/test_spark_s3_datasource.py

```diff
@@ -73,205 +65,68 @@ def spark_s3_datasource(s3_mock, s3_bucket: str) -> SparkS3Datasource:
-    spark_s3_datasource: SparkS3Datasource,
-):
-    asset = spark_s3_datasource.add_csv_asset(
-        name="csv_asset",
-        batching_regex=r"(?P<name>.+)_(.+)_(?P<price>\d{4})\.csv",
         header=True,
         infer_schema=True,
     )
-    options = asset.get_batch_parameters_keys()
-    assert options == (
-        "name",
-        "batch_request_param_2",
-        "price",
-        "path",
-    )
+    batching_regex = r"yellow_tripdata_sample_(?P<year>\d{4})-(?P<month>\d{2})\.csv"
+    batch_def = asset.add_batch_definition_monthly(name="batch def", regex=batching_regex)
+    options = asset.get_batch_parameters_keys(partitioner=batch_def.partitioner)
+    assert options == ("path", "year", "month")
 
 
-@pytest.mark.big
+@pytest.mark.unit
 def test_csv_asset_with_non_string_batching_regex_named_parameters(
     spark_s3_datasource: SparkS3Datasource,
 ):
     asset = spark_s3_datasource.add_csv_asset(
         name="csv_asset",
-        batching_regex=r"(.+)_(.+)_(?P<price>\d{4})\.csv",
         header=True,
         infer_schema=True,
     )
     with pytest.raises(ge_exceptions.InvalidBatchRequestError):
         # price is an int which will raise an error
-        asset.build_batch_request({"name": "alex", "timestamp": "1234567890", "price": 1300})
-
-
-@pytest.mark.big
-@pytest.mark.xfail(
-    reason="Accessing objects on moto.mock_s3 using Spark is not working (this test is conducted using Jupyter notebook manually)."  # noqa: E501
-)
-def test_get_batch_list_from_fully_specified_batch_request(
-    spark_s3_datasource: SparkS3Datasource,
-):
-    asset_specified_metadata = {"asset_level_metadata": "my_metadata"}
-    asset = spark_s3_datasource.add_csv_asset(
-        name="csv_asset",
-        batching_regex=r"(?P<name>.+)_(?P<timestamp>.+)_(?P<price>\d{4})\.csv",
-        header=True,
-        infer_schema=True,
-        batch_metadata=asset_specified_metadata,
-    )
+        asset.build_batch_request({"year": "2024", "month": 5})
 
-    request = asset.build_batch_request({"name": "alex", "timestamp": "20200819", "price": "1300"})
-    batches = asset.get_batch_list_from_batch_request(request)
-    assert len(batches) == 1
-    batch = batches[0]
-    assert batch.batch_request.datasource_name == spark_s3_datasource.name
-    assert batch.batch_request.data_asset_name == asset.name
-    assert batch.batch_request.options == {
-        "path": "alex_20200819_1300.csv",
-        "name": "alex",
-        "timestamp": "20200819",
-        "price": "1300",
-    }
-    assert batch.metadata == {
-        "path": "alex_20200819_1300.csv",
-        "name": "alex",
-        "timestamp": "20200819",
-        "price": "1300",
-        **asset_specified_metadata,
-    }
-    assert batch.id == "spark_s3_datasource-csv_asset-name_alex-timestamp_20200819-price_1300"
 
-    request = asset.build_batch_request({"name": "alex"})
-    batches = asset.get_batch_list_from_batch_request(request)
-    assert len(batches) == 2
```

Comment:
this test fails with an opaque `Py4JJavaError` which makes me wonder if this datasource is actually functional:
```
        request = asset.build_batch_request({"name": "alex", "timestamp": "20200819", "price": "1300"})
>       batches = asset.get_batch_list_from_batch_request(request)

test_spark_s3_datasource.py:214: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../great_expectations/datasource/fluent/data_asset/path/path_data_asset.py:144: in get_batch_list_from_batch_request
    data, markers = execution_engine.get_batch_data_and_markers(batch_spec=batch_spec)
../../../great_expectations/execution_engine/sparkdf_execution_engine.py:515: in get_batch_data_and_markers
    batch_data = reader_fn(path)
../../../env/lib/python3.10/site-packages/pyspark/sql/readwriter.py:727: in csv
    return self._df(self._jreader.csv(self._spark._sc._jvm.PythonUtils.toSeq(path)))
../../../env/lib/python3.10/site-packages/py4j/java_gateway.py:1322: in __call__
    return_value = get_return_value(
../../../env/lib/python3.10/site-packages/pyspark/errors/exceptions/captured.py:169: in deco
    return f(*a, **kw)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

answer = 'xro29'
gateway_client = <py4j.clientserver.JavaClient object at 0x2898baef0>
target_id = 'o26', name = 'csv'

    def get_return_value(answer, gateway_client, target_id=None, name=None):
        """Converts an answer received from the Java gateway into a Python object.
    
        For example, string representation of integers are converted to Python
        integer, string representation of objects are converted to JavaObject
        instances, etc.
    
        :param answer: the string returned by the Java gateway
        :param gateway_client: the gateway client used to communicate with the Java
            Gateway. Only necessary if the answer is a reference (e.g., object,
            list, map)
        :param target_id: the name of the object from which the answer comes from
            (e.g., *object1* in `object1.hello()`). Optional.
        :param name: the name of the member from which the answer comes from
            (e.g., *hello* in `object1.hello()`). Optional.
        """
        if is_error(answer)[0]:
            if len(answer) > 1:
                type = answer[1]
                value = OUTPUT_CONVERTER[type](answer[2:], gateway_client)
                if answer[1] == REFERENCE_TYPE:
>                   raise Py4JJavaError(
                        "An error occurred while calling {0}{1}{2}.\n".
                        format(target_id, ".", name), value)
E                   py4j.protocol.Py4JJavaError: An error occurred while calling o26.csv.
E                   : java.lang.RuntimeException: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
E                   	at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2688)
E                   	at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:3431)
E                   	at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3466)
E                   	at org.apache.hadoop.fs.FileSystem.access$300(FileSystem.java:174)
E                   	at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3574)
E                   	at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3521)
E                   	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:540)
E                   	at org.apache.hadoop.fs.Path.getFileSystem(Path.java:365)
E                   	at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$1(DataSource.scala:724)
E                   	at scala.collection.immutable.List.map(List.scala:293)
E                   	at org.apache.spark.sql.execution.datasources.DataSource$.checkAndGlobPathIfNecessary(DataSource.scala:722)
E                   	at org.apache.spark.sql.execution.datasources.DataSource.checkAndGlobPathIfNecessary(DataSource.scala:551)
E 
```

Reply:
I believe one needs to install an AWS provided jar to get spark to be able to read from s3.